### PR TITLE
SW-5780 Extract hardwired ID mapping helper function

### PR DESCRIPTION
**Pull request by sgrimm**
Commit 0ae61309 modified `AccessionImporterTest` to maintain a mapping between
hardwired IDs and the actual IDs generated by the database, so it could check
for the correct references between entities. Pull that logic out into a helper
function so it can be used in other tests.

### CODE REVIEW

**Comment by sgrimm** [comment]
* **#2271** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2271?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2270** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2270?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2269** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2269?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2268** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2268?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2267** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2267?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2266** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2266?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sgrimm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by sgrimm** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2269?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2270** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2270?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2269** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2269?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2268** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2268?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2267** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2267?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2266** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2266?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sgrimm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/test/kotlin/com/terraformation/backend/seedbank/db/AccessionImporterTest.kt [hunk 1]
```diff
@@ -390,15 +386,8 @@ internal class AccessionImporterTest : DatabaseTest(), RunsAsUser {
...
-      val mappedSpeciesIds =
-          actualSpecies
-              .mapIndexed { index, speciesRow -> SpeciesId(index + 1L) to speciesRow.id!! }
-              .toMap()
-      val mappedAccessionIds =
-          actualAccessions
-              .mapIndexed { index, accessionsRow -> AccessionId(index + 1L) to accessionsRow.id!! }
-              .toMap()
+      val mappedSpeciesIds = mapTo1IndexedIds(actualSpecies, ::SpeciesId, SpeciesRow::id)
+      val mappedAccessionIds = mapTo1IndexedIds(actualAccessions, ::AccessionId, AccessionsRow::id)
```
Comment (nickgraz) [inline src/test/kotlin/com/terraformation/backend/seedbank/db/AccessionImporterTest.kt:390]: 🔥

**Review by nickgraz — APPROVED** [review]
Excellent!

**Comment by sgrimm** [comment]
### Merge activity

* **Aug 12, 4:51 PM EDT**: @sgrimm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2269).
* **Aug 12, 5:17 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2269) rebased this pull request as part of a merge.
* **Aug 12, 5:27 PM EDT**: @sgrimm merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2269).

**Pull request merged**